packages/@ngtools/webpack/src/resource_loader.ts

```diff
@@ -1,23 +1,40 @@
-import {readFileSync} from 'fs';
 import * as vm from 'vm';
 import * as path from 'path';
 
 const NodeTemplatePlugin = require('webpack/lib/node/NodeTemplatePlugin');
 const NodeTargetPlugin = require('webpack/lib/node/NodeTargetPlugin');
 const LoaderTargetPlugin = require('webpack/lib/LoaderTargetPlugin');
 const SingleEntryPlugin = require('webpack/lib/SingleEntryPlugin');
+const loaderUtils = require('loader-utils');
 
 
+interface CompilationResult {
+  hash: string;
+  outputName: string;
+  source: string;
+  newSource?: string;
+}
 
 export class WebpackResourceLoader {
+  private _parentCompilation: any;
   private _context: string;
   private _uniqueId = 0;
+  private _cache: { [k: string]: CompilationResult } = {};
```

Comment:
Instead of a hash table, could you use a Map please?

Reply:
Done